homeassistant/components/home_connect/strings.json

```diff
@@ -188,6 +188,158 @@
         "name": "Internal light"
       }
     },
+    "select": {
+      "program": {
+        "name": "Program",
+        "state": {
+          "cleaning_cleanall": "Clean all",
+          "cleaning_cleanmap": "Clean map",
+          "basic_gohome": "Go home",
+          "beverage_ristretto": "Ristretto",
+          "beverage_espresso": "Espresso",
+          "beverage_espressodoppio": "Espresso doppio",
+          "beverage_coffee": "Coffee",
+          "beverage_xlcoffee": "XL coffee",
+          "beverage_caffegrande": "Caffe grande",
+          "beverage_espressomacchiato": "Espresso macchiato",
+          "beverage_cappuccino": "Cappuccino",
+          "beverage_lattemacchiato": "Latte macchiato",
+          "beverage_caffelatte": "Caffe latte",
+          "beverage_milkfroth": "Milk froth",
+          "beverage_warmmilk": "Warm milk",
+          "coffeeworld_kleinerbrauner": "Kleiner brauner",
+          "coffeeworld_grosserbrauner": "Grosser brauner",
+          "coffeeworld_verlaengerter": "Verlaengerter",
+          "coffeeworld_verlaengerterbraun": "Verlaengerter braun",
+          "coffeeworld_wienermelange": "Wiener melange",
+          "coffeeworld_flatwhite": "Flat white",
+          "coffeeworld_cortado": "Cortado",
+          "coffeeworld_cafecortado": "Cafe cortado",
+          "coffeeworld_cafeconleche": "Cafe con leche",
+          "coffeeworld_cafeaulait": "Cafe au lait",
+          "coffeeworld_doppio": "Doppio",
+          "coffeeworld_kaapi": "Kaapi",
+          "coffeeworld_koffieverkeerd": "Koffie verkeerd",
+          "coffeeworld_galao": "Galao",
+          "coffeeworld_garoto": "Garoto",
+          "coffeeworld_americano": "Americano",
+          "coffeeworld_redeye": "Red eye",
+          "coffeeworld_blackeye": "Black eye",
+          "coffeeworld_deadeye": "Dead eye",
+          "beverage_hotwater": "Hot water",
+          "prerinse": "Pre-rinse",
+          "auto1": "Auto 1",
+          "auto2": "Auto 2",
+          "auto3": "Auto 3",
+          "eco50": "Eco 50ºC",
+          "quick45": "Quick 45ºC",
+          "intensiv70": "Intensive 70ºC",
+          "normal65": "Normal 65ºC",
```

Comment:
Why do some now have a prefix and some don't? I think ideally we would have no prefix (I know we discussed this, but now that I think of it, people also use this for templates). If there is overlap without the prefixes, maybe the best way to go about it would be to assign a different translation key to every type.

Reply:
Can we transform the PascalCase to snake_case instead?

Reply:
AFAIK, Upper cases are not allowed at translation keys:
https://github.com/home-assistant/core/blob/10fdf819d381f96cf1efc3991c874fe1c146a89d/script/hassfest/translations.py#L23

Reply:
I mean lowercase snake_case.

Reply:
How do know that there's no overlap between the keys of we remove everything before program (including program)?

Reply:
Some commits ago, I used this was the way to do it:
https://github.com/home-assistant/core/blob/ffea342507baec57a7311c1de3d5b24181df6b4a/homeassistant/components/home_connect/strings.json#L225-L237

All was snake_case, and everything before program was there. But it required `stringcase` dependency because I used it for the transformation:
https://github.com/home-assistant/core/blob/ffea342507baec57a7311c1de3d5b24181df6b4a/homeassistant/components/home_connect/utils.py#L6-L8

Also because it resulted in very large translation keys, I decided to remove from everything before program and slugify the keys. I used `slugify` because I didn't know if there was any other tool to achieve snake_case.

Reply:
> How do know that there's no overlap between the keys of we remove everything before program (including program)?

Well, I don't really know if there is an appliance that is able to use programs with the same translation key and different functionality. If that exists, that would be a problem because the key of the map that translates between the slugified and splitted key and the real key is first one.

Reply:
For the string conversion, I think a regex replace could be used.

https://stackoverflow.com/a/1176023

Reply:
Unless we find some proof that we could simplify the keys, I suggest we keep the full keys to make sure there's no overlap.

Reply:
I don't understand this:

> the key of the map that translates between the slugified and splitted key and the real key is first one

Please clarify.

Reply:
> Please clarify.

hahaha sorry, too many "keys" 😅
I mean this:
https://github.com/home-assistant/core/blob/8959d6631071516003d5754c7b7f44f0e0fca3a9/homeassistant/components/home_connect/select.py#L72-L74
The formatted program is the key that can be overlapped, so having an overlap in the map will cause an issue in the functionality

Reply:
Right, thanks.